packages/cache/src/LocalSkipCacheStorage.ts

```diff
@@ -0,0 +1,39 @@
+import path from "path";
+import fs from "fs-extra";
+
+import { Logger } from "backfill-logger";
+
+import { CacheStorage } from "./CacheStorage";
+
+/**
+ * A CacheStorage that essentially just lets fetch return nothing locally, skipping cache, but verifies whether the hash is still correct based on the hasher algorithm
+ */
+export class LocalSkipCacheStorage extends CacheStorage {
+  constructor(
+    private internalCacheFolder: string,
+    logger: Logger,
+    cwd: string
+  ) {
+    super(logger, cwd);
+  }
+
+  protected getLocalCacheFolder(hash: string): string {
+    return path.resolve(this.cwd, this.internalCacheFolder, hash);
+  }
+
+  protected async _fetch(hash: string): Promise<boolean> {
+    const localCacheFolder = this.getLocalCacheFolder(hash);
+
+    if (!fs.pathExistsSync(localCacheFolder)) {
+      return false;
+    }
+
+    return true;
+  }
+
+  protected async _put(hash: string, _outputGlob: string[]): Promise<void> {
+    const localCacheFolder = this.getLocalCacheFolder(hash);
+
+    await fs.mkdirp(localCacheFolder);
```

Comment:
You could get into a scenario where you've reverted your changes and it should rebuild, but it skips it because it was built before. Maybe we should also remove old folders before adding a new one? Or maybe instead of using folders, we could store the hash in a text file?

Reply:
If you reverted the files, the hashes shouldn’t match, right? In that case, the hash object algorithm will catch that case. Maybe you’re thinking about a workflow that I’m not seeing. Can you list out the a potential flow ? 🙏

Reply:
Sure.

1.
`git checkout master`
`backfill -- yarn build` generates `lib/foo.js`
The hash generated is `123abc`, so we now get `node_modules/.cache/backfill/123abc/(empty)`

2.
`git checkout feature-branch`
`backfill -- yarn build` generates a different version of `lib/foo.js`
The hash generated is `456def`, so we now have `node_modules/.cache/backfill/{123abc|456def}/(empty)`

3.
`git checkout master`
`backfill -- yarn build` will generate the hash `123abc` and will find a cache hit and do a no-op.
But now `lib/foo.js` will be outdated.

Instead, if we generated `node_modules/.cache/backfill/hash` with the hash of the last built hash, we would not have had a cache hit in 3.

Does it make sense, or am I missing something?

Reply:
okay! so we use the same filename, but different content! got it!

Reply:
okay - fixed this one!